File: Code Forces/CPP/Sinking Ship.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
	int n;
	vector<string> rats;
	vector<string> woild;
	vector<string> men;
	string captain;

	cin>>n;
	for (int i = 0; i < n; ++i){
		string name, rank;
		cin>>name>>rank;
		if (rank == "rat")
			rats.push_back(name);
		if (rank == "woman" || rank == "child")
			woild.push_back(name);
		if (rank == "man")
			men.push_back(name);
		if (rank == "captain")
			captain = name;
	}

	for (int i = 0; i < (int)rats.size(); ++i)
		cout<<rats[i]<<endl;
	for (int i = 0; i < (int)woild.size(); ++i)
		cout<<woild[i]<<endl;
	for (int i = 0; i < (int)men.size(); ++i)
		cout<<men[i]<<endl;

	cout<<captain<<endl;
	return 0;
}

File: Code Forces/PY/Sinking Ship.py
n = int(input())

rats = []
woild = []
men = []
captain = ''

for i in range(n):
	name, rank = input().split()
	if rank == 'rat': rats.append(name)
	if rank == 'man': men.append(name)
	if rank == 'woman' or rank == 'child': woild.append(name)
	if rank == 'captain': captain = name


for rat in rats:print(rat)
for x in woild: print(x)
for man in men:print(man)
print(captain)